src/solver/conv_hip_implicit_gemm_bwd_data_xdlops.cpp

```diff
@@ -0,0 +1,391 @@
+        strides  = {ProblemInterpreter::GetAdjustedConvolutionStrideH(problem),
+                   ProblemInterpreter::GetAdjustedConvolutionStrideW(problem)};
+        dilation = {ProblemInterpreter::GetAdjustedConvolutionDilationH(problem),
+                    ProblemInterpreter::GetAdjustedConvolutionDilationW(problem)};
+        lPadding = {ProblemInterpreter::GetInputLeftPadH(problem),
+                    ProblemInterpreter::GetInputLeftPadW(problem)};
+        rPadding = {ProblemInterpreter::GetAdjustedInputRightPadH(problem),
+                    ProblemInterpreter::GetAdjustedInputRightPadW(problem)};
+    }
+    int N;
+    int K;
+    int C;
+    std::vector<int> input;
+    std::vector<int> output;
+    std::vector<int> filter;
+    std::vector<int> strides;
+    std::vector<int> dilation;
+    std::vector<int> lPadding;
+    std::vector<int> rPadding;
+};
+
+template <typename DataType>
+void PerformanceConfigHipImplicitGemmBwdXdlops::Init(const ProblemDescription& problem)
+{
+    const auto args      = CKArgsBwd{problem};
+    const auto conv_ptrs = DeviceOpBwdPtrs<DataType>::GetInstances();
+    assert(!conv_ptrs.empty());
+    this->total_size = conv_ptrs.size();
+    for(int i = 0; i < conv_ptrs.size(); i++)
+    {
+        auto argument_ptr = conv_ptrs[i]->MakeArgumentPointer(nullptr,
+                                                              nullptr,
+                                                              nullptr,
+                                                              args.N,
+                                                              args.K,
+                                                              args.C,
+                                                              args.input,
+                                                              args.filter,
+                                                              args.output,
+                                                              args.strides,
+                                                              args.dilation,
+                                                              args.lPadding,
+                                                              args.rPadding,
+                                                              {},
+                                                              {},
+                                                              {});
+        if(conv_ptrs[i]->IsSupportedArgument(argument_ptr.get()))
+        {
+            this->kernel_id = conv_ptrs[i]->GetTypeString();
+            break;
+        }
+        ++this->index;
+    }
+}
+
+template <typename DataType>
+bool PerformanceConfigHipImplicitGemmBwdXdlops::CheckIsSupportCKArgs(
+    const ProblemDescription& problem) const
+{
+    const auto args      = CKArgsBwd{problem};
+    const auto conv_ptrs = DeviceOpBwdPtrs<DataType>::GetInstances();
+    auto argument_ptr    = conv_ptrs[this->index]->MakeArgumentPointer(nullptr,
```

Comment:
Do we need to check if `this->index < conv_ptrs` ?

Reply:
@iq136boy Basically this is impossible because the value of `index` should be either validated (somewhere else) or produced from valid information read from perf-db.

@bghimireamd If in doubt, please add an assertion (not MIOPEN_THROW)